Welcome!

# **Payment Fraud Detection with Machine Learning**

> Fraud is one of the major issues accross industries such as banks, life insurance, health insurance, and many others... including the local coffee shop around the neighboorhood corner. But one such fraud that has been increasing a lot these days is fraud in making payments. For this ML project, the aim is proffer a solution model to fraud detection with machine learning, as well as making real-time decisions. ML model will be assessing individual customer behaviour as it happens. It’s constantly analyzing ‘normal’ customer activity, so when it spots an anomaly it can automatically block or flag a payment for review.

In [1]:
#importing Libaries
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix

In [2]:
#Loading Data
Fraud_Data = pd.read_csv('/content/Payment Fraud Data.csv')

In [3]:
#Taking a quick look at the data
Fraud_Data.head()

accountAgeDays  numItems  ...  paymentMethodAgeDays label
0              29         1  ...             28.204861     0
1             725         1  ...              0.000000     0
2             845         1  ...              0.000000     0
3             503         1  ...              0.000000     0
4            2000         1  ...              0.000000     0

[5 rows x 6 columns]

In [31]:
# Label Encoding Stay column in train dataset to numeric
from sklearn import preprocessing
lencoder = preprocessing.LabelEncoder()

Fraud_Data['paymentMethod'] = lencoder.fit_transform(Fraud_Data['paymentMethod'].astype('str'))


In [19]:
Fraud_Data.head()

accountAgeDays  numItems  ...  paymentMethodAgeDays  label
0              29         1  ...             28.204861      0
1             725         1  ...              0.000000      0
2             845         1  ...              0.000000      0
3             503         1  ...              0.000000      0
4            2000         1  ...              0.000000      0

[5 rows x 6 columns]

Checking data for possible data cleansing

In [33]:
Fraud_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39221 entries, 0 to 39220
Data columns (total 6 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   accountAgeDays        39221 non-null  int64  
 1   numItems              39221 non-null  int64  
 2   localTime             39221 non-null  float64
 3   paymentMethod         39221 non-null  int64  
 4   paymentMethodAgeDays  39221 non-null  float64
 5   label                 39221 non-null  int64  
dtypes: float64(2), int64(4)
memory usage: 1.8 MB


In [34]:
Fraud_Data.isnull().sum()

accountAgeDays          0
numItems                0
localTime               0
paymentMethod           0
paymentMethodAgeDays    0
label                   0
dtype: int64

Fortunately, the dataset is already well structured with no missing or any scope needing data cleaning in it.

Now, Let's dive into building our machine learning model.

## **Fraud Detection Model**

In [37]:
# Split dataset up into train and test sets
X = Fraud_Data.drop('label', axis =1 )
y = Fraud_Data['label']

X_train, X_test, y_train, y_test = train_test_split( X, y, test_size=0.33, random_state=17)

In [39]:
LogRegression = LogisticRegression()
LogReg_Predict = LogRegression.fit(X_train, y_train)

In [41]:
y_predition = LogReg_Predict.predict(X_test)
print(accuracy_score(y_predition, y_test))

1.0


From the accuracy score output, the fraud detection model gave an accuracy of 100 per cent by using the logistic regression algorithm.

## **Model Evaluation**

Now, let’s evaluate the model performance model. To evaluate, confusion matrix algorithm will be used to evaluate the model performance

In [42]:
# Compare test set predictions with ground truth labels
print(confusion_matrix(y_test, y_predition))

[[12753     0]
 [    0   190]]


From the above confusion matrix output,  of all the transaction in the dataset, 190 transactions are correctly recognized as fraud, and 12753 transactions are recognized as not fraudulent transactions.

In [ ]:
#End!